feat: bump corecrypto 1.0.0 rc.22 WPB-3384

netbe opened the pull request:
<!--do not remove this marker, its needed to replace info when ticket title is updated -->
<!--jira-description-action-hidden-marker-start-->

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-3384" title="WPB-3384" target="_blank"><img alt="Task" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10818?size=medium" />WPB-3384</a>  [iOS] Bump CoreCrypto to 1.0.0
  </td></table>
  <br />
 

<!--jira-description-action-hidden-marker-end-->
<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Upgrade CoreCrypto to v1.0.0-rc.22

Besides all the API changes the biggest change is that:
* most methods have turned async.
* byte arrays (`[Byte]`) in the API have been replaced by the `Data` type.

### Dependencies

Needs releases with:

- [ ] https://github.com/wireapp/wire-ios/pull/795

### Testing

- [x] I have added automated test to this contribution

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

netbe committed:
fix: Ciphersuite

typfel commented on wire-ios-share-engine/Sources/CoreCrypto/SharingSession+CoreCrypto.swift:
```diff
@@ -44,19 +44,25 @@ extension SharingSession {
                 )
 
-                let safeCoreCrypto = try SafeCoreCrypto(coreCryptoConfiguration: configuration)
+                // FIXME: CC
```
Surrounding method needs to become async?

typfel commented on wire-ios-share-engine/Sources/SharingSession.swift:
```diff
@@ -320,8 +320,8 @@ public class SharingSession {
 
-        coreDataStack.syncContext.performAndWait {
-            try? cryptoboxMigrationManager.completeMigration(syncContext: coreDataStack.syncContext)
+        Task {
```
`setUpCoreCryptoStack` should  be moved into this task once it's also async?

typfel commented on wire-ios-sync-engine/Source/SessionManager/SessionManager.swift:
```diff
@@ -894,49 +895,54 @@ public final class SessionManager: NSObject, SessionManagerType {
-                    fatalError("Failed to migrate data from CryptoBox to CoreCrypto keystore, error : \(error.localizedDescription)")
-                }
+            self.delegate?.sessionManagerWillMigrateAccount {
+                // FIXME: check
```
What should we check?

typfel commented on wire-ios-sync-engine/Source/Synchronization/ApplicationStatusDirectory.swift:
```diff
@@ -53,7 +53,9 @@ public final class ApplicationStatusDirectory: NSObject, ApplicationStatus {
         self.analytics = analytics
         self.teamInvitationStatus = TeamInvitationStatus()
-        self.operationStatus.isInBackground = application.applicationState == .background
+//        DispatchQueue.main.async { [application] in
```
Delete?

typfel commented on wire-ios-sync-engine/Source/Synchronization/Strategies/LabelDownstreamRequestStrategy.swift:
```diff
@@ -61,12 +61,17 @@ public class LabelDownstreamRequestStrategy: AbstractRequestStrategy, ZMEventCon
-    override public func nextRequestIfAllowed(for apiVersion: APIVersion) -> ZMTransportRequest? {
-        guard syncStatus.currentSyncPhase == .fetchingLabels || ZMUser.selfUser(in: managedObjectContext).needsToRefetchLabels else { return nil }
+    override public func nextRequestIfAllowed(for apiVersion: APIVersion) async -> ZMTransportRequest? {
+        var needsToRefetchLabels = false
+        managedObjectContext.performAndWait {
+            needsToRefetchLabels = ZMUser.selfUser(in: managedObjectContext).needsToRefetchLabels
+        }
```
```suggestion
        var needsToRefetchLabels = managedObjectContext.performAndWait {
            ZMUser.selfUser(in: managedObjectContext).needsToRefetchLabels
        }
```

typfel commented on wire-ios-sync-engine/Source/Synchronization/Strategies/UserClientEventConsumer.swift:
```diff
@@ -59,27 +59,29 @@ public class UserClientEventConsumer: NSObject, ZMEventConsumer {
         }
 
         let selfUser = ZMUser.selfUser(in: managedObjectContext)
+        // FIXME: CC - change function signature
```
Make surrounding function async

typfel replied:
Refactored in another PR.

typfel commented on wire-ios-sync-engine/Source/Synchronization/Strategies/UserClientRequestStrategy.swift:
```diff
@@ -396,7 +403,10 @@ public final class UserClientRequestStrategy: ZMObjectSyncStrategy, ZMObjectStra
         case .success:
             if let payload = response.payload?.asArray() as? [[String: AnyObject]] {
-                self.received(clients: payload)
+                // FIXME: CC check if ok async
```
It would not be OK if there are any request completion handlers attached to the request.

typfel commented on wire-ios-sync-engine/Source/Synchronization/ZMOperationLoop.m:
```diff
@@ -146,18 +150,28 @@ - (ZMTransportRequestGenerator)requestGenerator {
+        __block ZMTransportRequest *returnedRequest = nil;
+        NSAssert(![NSThread isMainThread], @"should not run on main Thread");
+        NSLog(@"Running on thread %@", NSThread.currentThread);
+        // semaphore to wait for each request to complete executed on private queue of OperationQueue
```
Why not execute the operation loop in a task?

typfel commented on wire-ios-sync-engine/Source/UserSession/ZMUserSession.swift:
```diff
@@ -361,6 +388,14 @@ public class ZMUserSession: NSObject {
+        operationLoop?.setupRequestAvailableObserver()
+        RequestAvailableNotification.notifyNewRequestsAvailable(nil)
+        syncStatus?.markAsReady()
+        // FIXME: CC - uncomment next line
```
What needs to be done before we can uncomment?

typfel commented on wire-ios/Wire-iOS/Sources/UserInterface/Helpers/UserInputRequest+LegalHold.swift:
```diff
@@ -30,7 +30,9 @@ extension UserType where Self: SelfLegalHoldSubject {
 
     func makeLegalHoldInputRequest(for request: LegalHoldRequest, cancellationHandler: @escaping () -> Void, inputHandler: @escaping (String?) -> Void) -> UserInputRequest {
-        let fingerprintString = self.fingerprint?.fingerprintStringWithSpaces ?? "<fingerprint unavailable>"
+        // FIXME: CC -  fix signature
```
Make function async

netbe commented on wire-ios-data-model/MessagingService.swift:
```diff
@@ -0,0 +1,120 @@
+        }
+    }
+
+    public func fetchAndAssignRemoteFingerprint(for userClient: UserClient) async {
```
you mean a simple fetch method that does not assign? As discussed it's probably only used by the UI so yes

typfel replied:
This code was refactored in another PR.

netbe commented on wire-ios-data-model/Source/MLS/CoreCryptoCallbacks.swift:
```diff
@@ -23,36 +23,35 @@ class CoreCryptoCallbacksImpl: CoreCryptoCallbacks {
     init() {}
 
-    func authorize(conversationId: ConversationId, clientId: ClientId) -> Bool {
+    func authorize(conversationId: Data, clientId: ClientId) -> Bool {
```
I think it's done previously on feat/mls, I need to update the branch with latest MLS changes

netbe commented on wire-ios-data-model/Source/MLS/SafeCoreCrypto.swift:
```diff
@@ -91,12 +105,35 @@ public class SafeCoreCrypto: SafeCoreCryptoProtocol {
+        return result
+    }
+
+    public func perform<T>(_ block: (CoreCryptoProtocol) throws -> T) async rethrows -> T {
```
there were a few CC methods that were non async if I recall

netbe committed:
make createGroup and performPendingJoins async

typfel committed:
refactor: make proteus session functions async

typfel committed:
refactor: make proteus session functions async

typfel committed:
test: update data model tests after async refactoring

typfel committed:
test: update data model tests after async refactoring

caldrian committed:
Merge branch 'develop' into feat/other-mls-async

caldrian committed:
fix: warnings

caldrian committed:
Merge remote-tracking branch 'github/feat/async-session' into feat/other-mls-async

caldrian committed:
fix: warnings

caldrian committed:
fix: more warnings

caldrian committed:
fix: tests in MLSServiceTests

typfel committed:
fix: message status response processing

typfel committed:
fix: message status response processing

typfel committed:
test: update request-strategy tests after async refactoring

typfel committed:
test: update request-strategy tests after async refactoring

caldrian committed:
chore: log unhandled errors in Task { ... }

caldrian committed:
fix: unit test in MLSService

caldrian committed:
Merge remote-tracking branch 'github/feat/async-session' into feat/other-mls-async

caldrian committed:
fix: unit test

caldrian committed:
fix: unit test in request strategy

caldrian committed:
fix build error in mocktransport

caldrian committed:
fix: build errors in integration tests

caldrian committed:
fix: some unit tests in sync engine

typfel committed:
fix: include all available async event consumers

typfel committed:
fix: use WaitingGroupTask so that we can wait for results in tests

typfel committed:
test: update sync engine tests after async refactoring

caldrian committed:
Merge remote-tracking branch 'github/feat/async-session' into feat/other-mls-async

# Conflicts:
#	wire-ios-data-model/Source/Model/User/ZMUser+LegalHoldRequest.swift
#	wire-ios-data-model/Tests/Source/Model/User/ZMUserLegalHoldTests.swift
#	wire-ios-data-model/Tests/Source/Model/UserClient/UserClientTests.swift
#	wire-ios-sync-engine/Source/Synchronization/Strategies/UserClientRequestStrategy.swift
#	wire-ios-sync-engine/Tests/Source/Integration/IntegrationTest+Encryption.swift
#	wire-ios-sync-engine/Tests/Source/Synchronization/Strategies/CallingRequestStrategyTests.swift
#	wire-ios-utilities/Source/Sequence+Async.swift

typfel committed:
delete commented out code

typfel committed:
fix: call completionHandler also in the async code path

typfel committed:
chore: disable test in test plan and a comment on what needs to be fixed

typfel committed:
chore: log error when deleting session

Co-authored-by: Christoph Aldrian <caldrian@users.noreply.github.com>

typfel committed:
fix: add missing await perform

typfel committed:
chore: update doc comment

typfel committed:
fix: make function private

typfel committed:
fix: accessing proteusSessionID on the wrong thread in test

typfel committed:
fix: revert hack

typfel committed:
chore: fix spelling

Co-authored-by: François Benaiteau <netbe@users.noreply.github.com>

typfel committed:
chore: add ticket reference

caldrian committed:
chore: add comment

caldrian committed:
Merge remote-tracking branch 'github/feat/async-session' into feat/other-mls-async

# Conflicts:
#	wire-ios-data-model/Source/Model/User/ZMUser+LegalHoldRequest.swift
#	wire-ios-data-model/Tests/Source/Model/UserClient/UserClientTests.swift

caldrian committed:
fix: remove redundant line

typfel committed:
chore: fix another core data concurrenncy issue

caldrian committed:
chore: remove asyncForEach

caldrian committed:
Merge remote-tracking branch 'github/feat/async-session' into feat/other-mls-async

caldrian committed:
fix: typo

caldrian committed:
Merge remote-tracking branch 'github/feat/async-session' into feat/other-mls-async

caldrian committed:
fix: address pr comments

netbe committed:
assert newMessage

netbe committed:
remove duplicate test

caldrian committed:
fix: remove redundant test

caldrian committed:
Merge remote-tracking branch 'github/develop' into feat/async-session

caldrian committed:
fix: tests

caldrian committed:
fix: undo changes

caldrian committed:
Update wire-ios-request-strategy/Tests/Helpers/MessagingTest+Encryption.swift

Co-authored-by: François Benaiteau <netbe@users.noreply.github.com>

caldrian committed:
fix: silent failing

caldrian committed:
Merge remote-tracking branch 'github/feat/async-session' into feat/other-mls-async

# Conflicts:
#	wire-ios-data-model/Tests/Source/Model/Messages/ZMClientMessageTests+Editing.swift

caldrian committed:
Merge remote-tracking branch 'github/develop' into feat/async-session

netbe committed:
chore: update CC version

caldrian committed:
fix: build error

caldrian committed:
fix: test

netbe committed:
chore: move MockCoreCrypto to DataModelSupport

netbe committed:
chore: remove duplicate MockEARServiceInterface

netbe committed:
remove duplicate MockCoreCrypto

netbe committed:
rename file to MockCoreCryptoProtocol

typfel committed:
fix: add missing implementation of enqueue() on mock transport

typfel committed:
Merge branch 'develop' into feat/async-session

netbe committed:
wip

netbe committed:
update header

netbe committed:
Update mockCoreCrypto via sourcery

netbe committed:
move files

netbe committed:
WIP

netbe committed:
add alias for ConversationId

netbe committed:
fix: rename mockCoreCrypto

typfel committed:
feat: async CoreCryptoProvider

typfel committed:
feat: async CoreCryptoProvider

typfel committed:
chore: use WaitingGroupTask in FetchingClientRequestStrategy

theswiftway commented on wire-ios-data-model/MessagingService.swift:
```diff
@@ -0,0 +1,120 @@
+
+import Foundation
+
+class MessagingService {
```
`Suggestion:` Add a `MessagingServiceInterface` protocol with functions that we intend to use from `ProteusProvider`, `MLSServiceInterface` and managedObjectContext

typfel replied:
Seems like this class has been deleted.

theswiftway commented on wire-ios-data-model/MessagingService.swift:
```diff
@@ -0,0 +1,120 @@
+
+public class FingerprintUseCase {
+
+    let messagingService: MessagingService
```
`Suggestion:` Use interface instead of concrete classes for dependency which would help us writing better unit tests.

theswiftway commented on wire-ios-data-model/Source/MLS/MLSActionExecutor.swift:
```diff
@@ -248,30 +252,31 @@ actor MLSActionExecutor: MLSActionExecutorProtocol {
-                let conversationInitBundle = try coreCrypto.perform { try $0.joinByExternalCommit(groupInfo: groupInfo.bytes,
-                                                                                                  customConfiguration: .init(keyRotationSpan: nil, wirePolicy: nil), credentialType: .basic) }
+            case .joinGroup(let publicGroupState):
+                let conversationInitBundle = try await coreCrypto.performAsync { try await $0.joinByExternalCommit(groupInfo: publicGroupState,
```
`Nitpick`: formatting could be better

typfel replied:
This code has since been refactored in another PR.

caldrian committed:
Merge remote-tracking branch 'github/feat/async-session' into feat/other-mls-async

# Conflicts:
#	wire-ios-data-model/Tests/MLS/MLSServiceTests.swift
#	wire-ios-mocktransport/Source/MockTransportSession.swift

caldrian committed:
Merge branch 'develop' into feat/async-session

caldrian committed:
Merge branch 'feat/async-session' into feat/other-mls-async

caldrian committed:
fix: issue from resolving merge conflicts

caldrian committed:
fix: build error in sync engine

netbe committed:
Merge branch 'develop' into feat/async-cc-provider

caldrian committed:
fix: CallingRequestStrategyTests

netbe committed:
update coreCryptoProvider access to be async

netbe committed:
Merge branch 'develop' into feat/async-cc-provider

netbe committed:
wip

netbe committed:
update methods and fix MLSDecryptionServiceTests

netbe committed:
turn tests async

caldrian committed:
Merge branch 'develop' into feat/other-mls-async

caldrian committed:
fix: SwiftLint violations

netbe committed:
rename method

netbe committed:
preprend call to coreCrypto.perform with await

netbe committed:
turn methods async

netbe committed:
Merge remote-tracking branch 'origin/feat/other-mls-async' into feat/async-cc-provider

netbe committed:
turn outOfSyncConversations async

netbe committed:
remove unused function

netbe committed:
merge changes upstream

netbe committed:
naive approach to just async all down to encrypt function

netbe committed:
wip passing context through

netbe committed:
refactor to avoid asyncCompactMap violation

netbe committed:
fix: issue with keyStore when not enabled in proteus by corecrypto

netbe committed:
fix: coredata multithread issue

netbe committed:
fix coredata threading

netbe committed:
fix: remove non needed functions

netbe committed:
wip

netbe committed:
fix tests

netbe committed:
wip

netbe committed:
wip

netbe committed:
reenable legacy tests

netbe committed:
move mock setup out of methods

caldrian committed:
Merge commit '4b61a1176a9839f65ad3c2f20bb8359690e58fea' into feat/async-cc-provider

caldrian committed:
chore: remove asyncForEach

netbe committed:
fix test

netbe committed:
fix tests

caldrian committed:
Merge branch 'develop' into feat/async-cc-provider

# Conflicts:
#	wire-ios-data-model/Source/MLS/MLSService.swift
#	wire-ios-data-model/Tests/MLS/MLSServiceTests.swift

netbe committed:
fix tests

caldrian committed:
fix: make more methods async

caldrian committed:
fix: some unit test

netbe committed:
Merge branch 'develop' into feat/refactor-encryptForTransport

netbe committed:
fix: compilation error making methods async

netbe committed:
remove categories to run unit tests individualy

netbe committed:
code review

netbe committed:
code review

netbe committed:
fix: crash on encryptionSession during tests

netbe committed:
fix: tests in datamodel

netbe committed:
fix: some coredata threading issue in tests

netbe committed:
Merge branch 'develop' into feat/async-cc-provider

netbe committed:
Merge branch 'feat/refactor-encryptForTransport' into feat/async-cc-provider

netbe committed:
fix thread issue

netbe committed:
fix other thread issue

netbe committed:
Merge branch 'feat/refactor-encryptForTransport' into feat/async-cc-provider

netbe committed:
fix alll datamodel tests

netbe committed:
fix: conversationExists

netbe committed:
fix: GetUserFingerprintUserCase

netbe committed:
temporary fix for createMLSClientIfNeeded

netbe committed:
restore prekey from ProteusService

netbe committed:
get lastPrekeyID async

netbe committed:
update mock

netbe committed:
fix coredata threading violation

netbe committed:
disable legalhold for now - to be reverted

netbe committed:
try waiting task

caldrian committed:
Merge branch 'develop' into feat/refactor-encryptForTransport

caldrian committed:
Revert "Revert "feat: turn MLS encryption and decryption service async - WPB-5941 (#797)""

This reverts commit 39cc459c6da12bbad1cf55ec887e7d9283c3df2d.

caldrian committed:
chore: remove async CryptoBox methods

caldrian committed:
fix: context.performAndWait

caldrian committed:
fix: access on wrong queue

caldrian committed:
Merge branch 'develop' into feat/refactor-encryptForTransport

caldrian committed:
fix: context.performAndWait

caldrian committed:
Merge branch 'feat/refactor-encryptForTransport' of github.com:wireapp/wire-ios into feat/async-cc-provider

netbe committed:
Merge branch 'develop' into feat/refactor-encryptForTransport

netbe committed:
Update wire-ios-data-model/Source/Proteus/ProteusProvider.swift

Co-authored-by: Arun <arunbalakrishnan89@gmail.com>

netbe committed:
fix: notifications showing in background

netbe committed:
feat: make systemLogger messages public on debug

netbe committed:
prefix methods with legacy to avoid wrong method to be called

Commit:
Merge branch 'feat/refactor-encryptForTransport' into feat/async-cc-provider

netbe committed:
merge cc async setup branch

netbe committed:
wip

netbe committed:
fix types after update

netbe committed:
update mock

netbe committed:
update part of the mocks in datamodel

typfel committed:
Merge branch 'develop' into feat/async-cc-provider

typfel committed:
fix: wasn't setting initialisatingMls to true

Co-authored-by: Christoph Aldrian <caldrian@users.noreply.github.com>

typfel committed:
chore: rename to initialisatingMLS

typfel committed:
refactor: use already existing test helper method

typfel committed:
refactor: move if statement outside Task

typfel committed:
refactor: consolidate usage of error test helper

typfel committed:
refactor: make legalhold fingerprint async

typfel committed:
Merge branch 'feat/async-cc-provider' into feat/bump-corecrypto-1.0-rc-WPB-3384

typfel commented on wire-ios-data-model/Source/MLS/SafeCoreCrypto.swift:
```diff
@@ -116,18 +132,36 @@ public class SafeCoreCrypto: SafeCoreCryptoProtocol {
     }
 
-    private func restoreFromDisk() {
+    private func restoreFromDisk() async {
```
Why not let it throw?

typfel replied:
Ok, because the surrounding function doesn't throw atm. Let's deal with this later.

typfel commented on wire-ios-data-model/Source/Model/Message/ZMClientMessage+Encryption.swift:
```diff
@@ -355,14 +351,14 @@ extension GenericMessage {
         // TODO: get core crypto file lock
```
File lock is taken, we can delete this TODO

typfel committed:
chore: delete outdated comment

typfel commented on wire-ios-data-model/Source/Model/Message/ZMClientMessage+Encryption.swift:
```diff
@@ -386,16 +382,23 @@ extension GenericMessage {
+            } catch {
+                // FIXME: CC - handle error here
+                WireLogger.proteus.error("Error while rollback of session because payload too big: \(error)")
+
```
Refactored in another PR.

typfel commented on wire-ios-data-model/Source/Model/UserClient/UserClient.swift:
```diff
@@ -440,48 +453,58 @@ public extension UserClient {
+//                    }
+//
+//                }
+//            }
```
Refactored in another PR.

typfel commented on wire-ios-cryptobox/WireCryptobox/EncryptionContext.swift:
```diff
@@ -136,7 +136,7 @@ extension EncryptionContext {
             )
         }
         performCount += 1
```
it's there's to support nested `perform` blocks

typfel committed:
fix: protect against initialising MLS twice

typfel committed:
Merge branch 'feat/async-cc-provider' into feat/bump-corecrypto-1.0-rc-WPB-3384

typfel committed:
chore: upgrade CoreCrypto to v1.0.0-rc.22

typfel committed:
Merge branch 'develop' into feat/async-cc-provider

typfel committed:
Merge branch 'feat/async-cc-provider' into feat/bump-corecrypto-1.0-rc-WPB-3384

caldrian committed:
fix: undo .swiftlint.yml changes

theswiftway commented on wire-ios-data-model/Source/MLS/CoreCryptoProvider.swift:
```diff
@@ -233,7 +234,7 @@ public actor CoreCryptoProvider: CoreCryptoProviderProtocol {
 
         WireLogger.mls.info("generating ed25519 public key")
-        let keyBytes = try await coreCrypto.perform { try $0.clientPublicKey(ciphersuite: defaultCipherSuite.rawValue) }
+        let keyBytes = try await coreCrypto.perform { try await $0.clientPublicKey(ciphersuite: CiphersuiteName.default.rawValue) }
```
Would making the `CiphersuiteName` be part of the initialiser to make it configurable?

typfel replied:
Yes, that would probably make sense when we need that in the future, maybe for e2ei?

typfel replied:
Let's deal with this when we actually need configurable ciphersuites, I don't want to guess on what API we will need in the future.

theswiftway commented on wire-ios-data-model/Source/MLS/SafeCoreCrypto.swift:
```diff
@@ -42,43 +67,35 @@ public class SafeCoreCrypto: SafeCoreCryptoProtocol {
-            key: key,
-            ciphersuites: [defaultCipherSuite.rawValue]
-        )
+    public convenience init(path: String, key: String) async throws {
```
Would making the CiphersuiteName be part of the initialiser to make it configurable?

typfel replied:
Hmm actually the ciphersuite is irrelevant in this case we are only initialising proteus here and it doesn't care about the ciphersuite.

typfel replied:
I'll add comment

theswiftway commented on wire-ios-data-model/Source/MLS/SafeCoreCrypto.swift:
```diff
@@ -42,43 +67,35 @@ public class SafeCoreCrypto: SafeCoreCryptoProtocol {
-                               ciphersuites: [defaultCipherSuite.rawValue])
+        // TODO: wait for fix see cyphersuite cyphersuiteName
+        try await coreCrypto.mlsInit(clientId: clientIdBytes,
+                                     ciphersuites: [CiphersuiteName.default.rawValue],
```
We could avoid the problem using different configurations for cipherSuites at different places by fetching it from a class variable instead of setting it here

typfel replied:
Hmm, yeah let review this. maybe this whole thing actually move to `CoreCryptoProvider`.

theswiftway commented:
It would be nice to add the reason for changing the byte arrays ([Byte]) in the API into the Data type.

typfel commented:
> It would be nice to add the reason for changing the byte arrays ([Byte]) in the API into the Data type.

The change originates the from CoreCrypto, previously their API was using byte arrays because that was their Swift binding generator was producing them. That eventually that got fixed and the bindings now use the `Data` struct instead which is more swifty.

caldrian committed:
fix: generation of MockCoreCryptoProtocol

caldrian commented on wire-ios-data-model/Source/MLS/MLSService.swift:
```diff
@@ -423,14 +423,15 @@ public final class MLSService: MLSServiceInterface {
                 externalSenders: backendPublicKeys.ed25519Keys,
-                custom: .init(keyRotationSpan: nil, wirePolicy: nil)
+                custom: .init(keyRotationSpan: nil, wirePolicy: nil),
+                perDomainTrustAnchors: [PerDomainTrustAnchor.init(domainName: "", intermediateCertificateChain: "")] // TODO: review this
```
What does this TODO mean?
I assume leaving it as is has the same effect as removing it :)

typfel replied:
I didn't add this :) but trust anchors is something which comes with end2end identify. I think we can remove it.

typfel replied:
Remove it and now also don't configure any trust anchors at all.

typfel committed:
chore: remove unused code

- didInitializeMLS is uncessary since protect against this in CoreCryptoProvider
- remove unused convenience init

caldrian commented on wire-ios-data-model/Source/MLS/SafeCoreCrypto.swift:
```diff
@@ -116,18 +133,36 @@ public class SafeCoreCrypto: SafeCoreCryptoProtocol {
-    // TODO: Remove when async core crypto is supported
-    public func perform<T>(_ block: (WireCoreCrypto.CoreCryptoProtocol) async throws -> T) async rethrows -> T {
-        try await block(coreCrypto)
+    public func perform<T>(_ block: (WireCoreCrypto.CoreCryptoProtocol) throws -> T) async rethrows -> T {
```
This method can be removed, right?
It won't be an issue if a provided block contains no `await` and is called with
```
try await block()
```

So the `perform` method above should be sufficient.

typfel replied:
I think you're right, let's remove it.

typfel replied:
Done

typfel committed:
chore: clean up

Co-authored-by: David-Henner <david.henner@wire.com>

caldrian committed:
fix: add missing context.perform

typfel committed:
Merge branch 'feat/async-cc-provider' into feat/bump-corecrypto-1.0-rc-WPB-3384

typfel committed:
chore: don't configure any trust anchors at all for now

typfel committed:
chore: delete non-async perform

typfel committed:
Merge branch 'develop' into feat/async-cc-provider

typfel committed:
Merge branch 'feat/async-cc-provider' into feat/bump-corecrypto-1.0-rc-WPB-3384

typfel committed:
fix: flaky test not waiting for task to complete

typfel committed:
Merge branch 'develop' into feat/async-cc-provider

typfel committed:
chore: remove now obsolete test

typfel committed:
Merge branch 'feat/async-cc-provider' into feat/bump-corecrypto-1.0-rc-WPB-3384

caldrian committed:
Merge remote-tracking branch 'github/develop' into feat/async-cc-provider

caldrian committed:
Merge branch 'feat/async-cc-provider' into feat/bump-corecrypto-1.0-rc-WPB-3384

# Conflicts:
#	Cartfile
